scripts/validate_staging_patches.py

```diff
@@ -0,0 +1,47 @@
+#! /usr/bin/env python
+
+from pathlib import Path
+import subprocess
+import sys
+
+
+def main():
+    # Find all staging patches
+    patch_files = list(Path(__file__).parent.parent.rglob("**/staging/**/*.y*ml"))
+
+    for patch_file in patch_files:
+
+        # Go to root of namespace directory (i.e. 'k8s/gitlab' for gitlab namespace yamls)
+        current_dir = patch_file.parent
+        while current_dir.parent.name != "k8s":
+            current_dir = current_dir.parent
+
+        # Run the gitops-patch.py script on each yaml in the namespace. If one of them succeeds,
+        # assume it is valid.
```

Comment:
Just for my own understanding, why is it that the success of one file constitutes the success of the entire directory?

Reply:
The original idea was that if the staging yaml applied successfully to a single "regular" yaml, then it's valid. Which may not necessarily be true, but it gives a non-zero amount of confidence that all staging patches are valid. 
However, I've rewritten the script to explicitly apply each staging patch to their intended yaml, so this should now work 100% of the time.